streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -114,6 +115,12 @@ void addNewTask(final T task) {
         return readyPartitions;
     }
 
+    void initializeTopologyForRestored() {
+        for (final Map.Entry<TaskId, T> taskIdTEntry : restoring.entrySet()) {
```

Comment:
Hmm, is this correct? `restoring` contains those tasks that are still restoring not not finished. 

Should we just call `initializeTopology()` in the `transitionToRunning` function? (This way we would not need to add the line 106 above).

Reply:
Ack, good catch.